src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,28 +24,153 @@
+            if (countLimit >= numStudentsToAdd) {
+                return;
+            }
+            if (!testData.students.containsKey(key)) {
+                testData.students.put(key, testDataMax.students.get(key));
+                countLimit++;
+            }
+        }
+    }
+
+    private void decreaseNumOfStudents(int remainingNumOfStudents) {
+        Set<String> studentsNames = testData.students.keySet();
+        for (String key : studentsNames) {
+            if (testData.students.size() <= remainingNumOfStudents) {
+                return;
+            }
+            testData.students.remove(key);
+        }
+    }
+
+    private void increaseNumOfQuestions(int numQuestionsToAdd) {
+        int countLimit = 0;
+        for (String key : testDataMax.feedbackQuestions.keySet()) {
+            if (countLimit >= numQuestionsToAdd) {
+                return;
+            }
+            if (!testData.feedbackQuestions.containsKey(key)) {
+                testData.feedbackQuestions.put(key, testDataMax.feedbackQuestions.get(key));
+                countLimit++;
+            }
+        }
+    }
+
+    private void decreaseNumOfQuestions(int remainingNumOfQuestions) {
+        Set<String> questionsKeys = testData.feedbackQuestions.keySet();
+        for (String key : questionsKeys) {
+            if (testData.feedbackQuestions.size() <= remainingNumOfQuestions) {
+                return;
+            }
+            testData.feedbackQuestions.remove(key);
+        }
+    }
+
+    private void updateFeedbackResponses() {
+        // obtain set of e-mails for current set of students
+        Set<String> studentsEmails = new HashSet<>(testData.students.size());
+        for (StudentAttributes student : testData.students.values()) {
+            studentsEmails.add(student.email);
+        }
+
+        // obtain set of keys for current Map of questions
+        Set<Integer> questionsNumbers = new HashSet<>(testData.feedbackQuestions.size());
+        for (FeedbackQuestionAttributes question : testData.feedbackQuestions.values()) {
+            questionsNumbers.add(question.questionNumber);
+        }
+
+        // obtain set of keys for all feedback responses
+        Set<String> feedbackResponsesKeys = testDataMax.feedbackResponses.keySet();
+        String giver;
+        String recipient;
+        String questionId;
+
+        // check every feedback response if it should be included to test or excluded from it
+        for (String key : feedbackResponsesKeys) {
+            giver = testDataMax.feedbackResponses.get(key).giver;
+            recipient = testDataMax.feedbackResponses.get(key).recipient;
+            questionId = testDataMax.feedbackResponses.get(key).feedbackQuestionId;
+
+            if (studentsEmails.contains(giver)
+                    && studentsEmails.contains(recipient)
+                    && questionsNumbers.contains(questionId)) {
+                if (!testData.feedbackResponses.containsKey(key)) {
```

Comment:
Consider combining these two if statements. You can abstract the condition out if it becomes over long.